### PR TITLE
feat: Allow to use references to existing instances

### PR DESCRIPTION
**Pull request by andreainnocenti**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -41,8 +87,15 @@ module "cos" {
...
 
+data "ibm_resource_instance" "existing_studio_instance" {
+  provider   = ibm.deployer
+  count      = var.existing_studio_instance != null ? 1 : 0
+  identifier = var.existing_studio_instance
+}
+
 resource "ibm_resource_instance" "studio_instance" {
   provider          = ibm.deployer
+  count             = var.existing_studio_instance != null ? 0 : 1
```
Comment (in-1911) [inline main.tf:98]: Could you add `moved { .... }` block to avoid destructive behavior on an upgrade?

Comment (andreainnocenti) [reply main.tf:98]: I didn't rename any resource, what do you mean with `destructive behavior on an upgrade`? According to https://developer.hashicorp.com/terraform/language/moved:

```
The moved block programmatically changes the address of a resource
```

I don't think this is the case.

**Review by mkrudele — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (in-1911) [reply main.tf:98]: The address changes from `ibm_resource_instance.studio_instance` to  `ibm_resource_instance.studio_instance[0]`. That will trigger destroy/create.

Comment (in-1911) [reply main.tf:98]: See https://developer.hashicorp.com/terraform/language/modules/develop/refactoring#enabling-count-or-for_each-for-a-resource

**Review by in-1911 — APPROVED** [review]

**Pull request merged**